src/plugins/controls/public/range_slider/embeddable/range_slider_embeddable.tsx

```diff
@@ -228,39 +238,21 @@ export class RangeSliderEmbeddable
   };
 
   private runRangeSliderQuery = async () => {
-    this.dispatch.setLoading(true);
-
     const { dataView, field } = await this.getCurrentDataViewAndField();
     if (!dataView || !field) return;
 
-    const embeddableInput = this.getInput();
-    const { ignoreParentSettings, timeRange: globalTimeRange, timeslice } = embeddableInput;
-    let { filters = [] } = embeddableInput;
-
-    const timeRange =
-      timeslice !== undefined
-        ? {
-            from: new Date(timeslice[0]).toISOString(),
-            to: new Date(timeslice[1]).toISOString(),
-            mode: 'absolute' as 'absolute',
-          }
-        : globalTimeRange;
-    if (!ignoreParentSettings?.ignoreTimerange && timeRange) {
-      const timeFilter = this.dataService.timefilter.createFilter(dataView, timeRange);
-      if (timeFilter) {
-        filters = filters.concat(timeFilter);
-      }
-    }
+    this.dispatch.setLoading(true);
```

Comment:
```suggestion
```

We don't need this now that the subscriptions are setting the loading state :eyes:

Reply:
Hm, we also call `runRangeSliderQuery` from the public `reload` method. So we could also set the loading state during `reload` or keep setting it in the `runRangeSliderQuery` method. What do you think @Heenawter?

Reply:
Oh, good call! I think maybe having the `setLoading` calls in the `reload` method is slightly cleaner than hiding them in the `runRangeSliderQuery`? What do you think?

Also, with the current code, for the `dataFetchPipe` - we are setting the loading state to `false` **early** because we set it to the false at the end of `runRangeSliderQuery`, but then we aren't setting it to `true` again for `runValidations`.

Reply:
Fixed in https://github.com/elastic/kibana/pull/174201/commits/be3d10f012079cab40bf1b16f5f53a2a0aac48f0